projects/ExtraRepository.cmake

```diff
@@ -0,0 +1,33 @@
+foreach(extra_repository ${EXTRA_REPOSITORIES})
```

Comment:
What I mean is that we have a separate directory with project files similar to the project files in this repo that are also added during the build step. This gives us far more flexibility as to what we actually want to do with those extra bits instead of limiting them to a git url and some cmake options. Something along the line of "for each *.cmake file in BUILD_EXTRA_PROJECTS: include(file); done". This makes it possible to have extra repositories version controlled and greatly simplifies this code.